content/_includes/contact.macros.njk

```diff
@@ -144,6 +144,20 @@ params:
           ) -}}
           {{- context if not hide_details -}}
         </li>
+      {% elif site == 'mastodon' %}
+        {%- set text = ['@', id.user, '@', id.server] | join -%}
+        {%- set context %}<span class="sr-only"><i>on</i> Mastodon</span>{% endset -%}
+        {%- set text = text + context if not details -%}
+        <li>
+          {{- utility.icon_link(
+            icon='social/mastodon',
+            text=text | safe,
+            url=['https://', id.server, '/@', id.user] | join,
+            class='u-url',
+            attrs={'rel': rel} if rel else {}
+          ) -}}
+          {{- context if details -}}
```

Comment:
I notice we never use `details=true` with this macro, so we could probably remove it and simplify this.